qemu/tests/chardev_tls_encryption.py

```diff
@@ -0,0 +1,95 @@
+import time
+import aexpect
+
+from avocado.utils import process
+
+from virttest import utils_misc
+from virttest import env_process
+from virttest import error_context
+
+from provider.chardev_utils import setup_certs
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test native TLS encryption on chardev TCP transports
+    Scenario 1:
+        a. Run gnutls server
+        b. Launch QEMU with a serial port as TLS client
+        c. Check the server endpoint output
+    Scenario 2:
+        a. Launch QEMU with a serial port as TLS server
+        b. Run gnutls client to connect TLS server
+        c. Check the client endpoint output
+    Scenario 3:
+        a. Launch QEMU with a serial port as TLS server
+        b. Execute 'cat /dev/ttyS0' in guest which boot from step 1
+        c. Launch QEMU with a serial port as TLS client
+        d. Check the output of step b
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    try:
+        setup_certs(params)
+        cert_dir = params["cert_dir"]
+        hostname = process.run('hostname', ignore_status=False, shell=True,
+                               verbose=True).stdout_text.strip()
+        port = str(utils_misc.find_free_ports(5000, 9999, 1, hostname)[0])
+
+        # Scenario 1
+        gnutls_cmd_server = params.get("gnutls_cmd_server")
+        if gnutls_cmd_server:
+            gnutls_cmd_server = gnutls_cmd_server % (cert_dir, port)
+            params["extra_params"] = params["extra_params"] % (
+                cert_dir, hostname, port)
+            tls_server = aexpect.run_bg(gnutls_cmd_server)
+            params['start_vm'] = 'yes'
+            vm_name = params['main_vm']
+            env_process.preprocess_vm(test, params, env, vm_name)
+            time.sleep(15)
+            if "Channel binding 'tls-unique'" not in tls_server.get_output():
+                test.fail("tls server can't connect client succssfully.")
+
+        # Scenario 2
+        gnutls_cmd_client = params.get("gnutls_cmd_client")
+        if gnutls_cmd_client:
+            gnutls_cmd_client = gnutls_cmd_client % (cert_dir, port, hostname)
+            params["extra_params"] = params["extra_params"] % (
+                cert_dir, hostname, port)
+            params['start_vm'] = 'yes'
+            vm_name = params['main_vm']
+            env_process.preprocess_vm(test, params, env, vm_name)
+            tls_client = aexpect.run_bg(gnutls_cmd_client)
+            time.sleep(15)
+            if "Simple Client Mode:" not in tls_client.get_output():
```

Comment:
Could you have a try:
```
if not utils_misc.wait_for(lambda: "Simple Client Mode:" in  tls_client.get_output(), first=5, timeout=**):                  
    test.fail('****')  
```

Reply:
> Could you have a try:
> 
> ```
> if not utils_misc.wait_for(lambda: "Simple Client Mode:" in  tls_client.get_output(), first=5, timeout=**):                  
>     test.fail('****')  
> ```

@yanan-fu Thanks and update.

Test pass.
 (1/3) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.1.x86_64.io-github-autotest-qemu.chardev_tls_encryption.host_to_guest: PASS (20.46 s)
 (2/3) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.1.x86_64.io-github-autotest-qemu.chardev_tls_encryption.guest_to_host: PASS (20.10 s)
 (3/3) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.1.x86_64.io-github-autotest-qemu.chardev_tls_encryption.guest_to_guest: PASS (77.30 s)